### PR TITLE
Support translation of template fields

### PR DESCRIPTION
**Pull request by AntoineAugusti**
Related to https://github.com/cds-snc/notification-api/issues/850

It's possible to translate things in Jinja templates by adding brackets. This PR adds the possibility to translate placeholders.

The idea is to add brackets so that they'll be translated at the rendering phase.

### CODE REVIEW

File: notifications_utils/field.py [hunk 1]
```diff
@@ -57,7 +57,8 @@ class Field:
     placeholder_tag = "<span class='placeholder'>(({}))</span>"
     conditional_placeholder_tag = "<span class='placeholder-conditional'>(({}??</span>{}))"
     placeholder_tag_no_brackets = "<span class='placeholder-no-brackets'>{}</span>"
-    placeholder_tag_redacted = "<span class='placeholder-redacted'>hidden</span>"
+    placeholder_tag_no_brackets_translated = "<span class='placeholder-no-brackets'>[{}]</span>"
+    placeholder_tag_redacted = "<span class='placeholder-redacted'>[hidden]</span>"
```
Comment (AntoineAugusti) [inline notifications_utils/field.py:61]: Added brackets to this magic value to enable translation on the admin

File: notifications_utils/version.py [hunk 2]
```diff
@@ -1,2 +1,2 @@
-__version__ = '43.0.2'
+__version__ = '43.0.4'
```
Comment (AntoineAugusti) [inline notifications_utils/version.py:1]: Assuming https://github.com/cds-snc/notification-utils/pull/61, version 43.0.3 will be merged before

**Review by brdunfield — APPROVED** [review]
My only comment is that it looks like with_brackets is no longer being used (as the only other instance of it being set to False is in a Letter template, and we do not do letters).

If that's the case, we might want to consider removing it to simplify the code. But everything here looks good!

**Comment by AntoineAugusti** [comment]
Thanks for taking a look! I thought about removing this option and did not go for it but it seems like a good idea to do extra cleaning while I'm at it 👌 I'll revert to draft, work on it and request another review from you.

**Comment by AntoineAugusti** [comment]
Removed the `with_brackets_ attribute. We'll need to translate address lines if we want to sender letters in the future

**Review by brdunfield — APPROVED** [review]
Still good to go, thanks for cleaning that up! 👍

**Pull request merged**